src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/RecentArrivalsService.kt

```diff
@@ -4,33 +4,35 @@ import org.springframework.data.domain.Page
 import org.springframework.data.domain.PageImpl
 import org.springframework.data.domain.PageRequest
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search.SearchByNameAndPrisonNumber
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search.Searcher
 import uk.gov.justice.digital.hmpps.welcometoprison.model.prison.PrisonApiClient
 import java.time.LocalDate
 import java.time.LocalTime
 
 @Service
 class RecentArrivalsService(
-  private val prisonApiClient: PrisonApiClient
+  private val prisonApiClient: PrisonApiClient,
+  private val searcher: Searcher<String, RecentArrival> = Searcher(SearchByNameAndPrisonNumber())
 ) {
+
   fun getArrivals(
     prisonId: String,
-    fromDate: LocalDate,
-    toDate: LocalDate,
-    pageSize: Int,
-    page: Int
+    dateRange: Pair<LocalDate, LocalDate>,
+    page: PageRequest,
+    query: String? = null
   ): Page<RecentArrival> {
```

Comment:
didn't want to add another parameter as list was getting long so moved to use a pair and page object